### ISSUE TITLE
handle map with union Map<string, DU>

### ISSUE THREAD

**Issue by AviAvni** [Zaid-Ajaj/LiteDB.FSharp#4]
when tring to serialize map that the value is Descriminated Union there is null reference exception

**Comment by Zaid-Ajaj** [comment]
Can you elaborate with more details?

**Comment by AviAvni** [comment]
Try to serialize
```
type Value = Num of int | String of string
type Entity = { Id:string: Properties:map<string, Value> }
```
I get exception in Bson.rewriteKey

**Comment by AviAvni** [comment]
Great can you push the fix to nuget?

**Comment by Zaid-Ajaj** [comment]
Thanks for the details. The `rewriteKey` function will now ignore maps because these are not nested types. 

I already pushed the fix to nuget, version 0.8.1 is now being indexed :smile:

**Comment by AviAvni** [comment]
Thank you very much great job it's help me a lot
